### DISCUSSION TITLE
Can I map <M-c> to COPY with kMonad?

### DISCUSSION THREAD

**Discussion by leira** [kmonad/kmonad#461] [Q&A] [answered]
I use a Linux box and a MacBook Pro. I really hate to deal with the keyboard shortcut differences. With my current key map with KMonad, it is easier to press <M-c> than <C-c>, so I plan to map <M-c>, <M-x> and <M-v> to COPY, CUT and PASTE. Is it possible to do with KMonad? Or is it outside of the scope of KMonad? If so, what tools do such mapping, and work well with KMonad?

**Comment by stephane-klein** [comment]
@leira did you find an answer to your question?

**Comment by slotThe** [comment]
Depending on your system, you can try tools that interact with the clipboard (e.g. `xclip`).  I don't know how to do this on Apple devices.

**Comment by MuhammedZakir** [comment, marked as answer]
Why not remap keys to actual shortcut of those actions? That is, if you want to use Mac-style shortcut, then map `lmet`+{`c`, `x`, `v`} to `lctl`+{`c`, `x`, `v`}. Have you tried that?

**Reply by stephane-klein** [reply]
@MuhammedZakir I'm not familiar with kmonad syntax, can you paste a "complet" example ?🙏

**Reply by MuhammedZakir** [reply]
Note: This is not something I like as this makes things more complicated than needed. Unfortunately, until the refactoring is complete, we will have to make do with these.

Edit: I thought of a better, but hackier solution. I haven't tested it, so don't know whether it will work.

```clojure
(defcfg
  <your config>
)

(defsrc
  z  x  c  v
  lctl  lmet
)


(defalias
  lctl (around lctl (layer-toggle left-control))
)

(deflayer main
  z  x  c  v
  @lctl  lmet
)

(defalias
  lctlR (release-only lctl)
  lctlP (press-only lctl)
  x (tap-macro-release @lctlR M-x @lctlP)
  c (tap-macro-release @lctlR M-c @lctlP)
  v (tap-macro-release @lctlR M-v @lctlP)
)

(deflayer left-control
  _  M-x  M-c  M-v
  _  _
)
```

**Reply by leira** [reply]
This is definitely one solution. Rather than map `<M-c>` to `<C-c>`, is it possible to map it directly to [KeyCopy](https://github.com/kmonad/kmonad/blob/a8f2178f9d5f52bbb72cfc2bcc19597f6af02133/src/KMonad/Keyboard/Keycode.hs#L181)?

**Reply by markstos** [reply]
> Why not remap keys to actual shortcut of those actions? 
Besides that the copy/paste keys differ between Linux and Mac, they also differ between terminals and most other apps. Since termains use "Control-C" to mean "Interrupt", in a terminal "Control-Shift-C" is used instead. But several terminals (alacritty and foot for sure) understand the "Copy" keycode.

So as long as all the apps you use understand the Copy and Paste and keycodes, you can use these keycodes for universal copy and paste across Mac, Linux and various apps. The downside is that all apps doesn't support these yet. For example, the QT framework has an issue pending to support the codes there, but it looks like they will. 

So far, I've app I've asked to support the Copy and Paste key codes has agreed. If you are using an app that allows you to define your own keybindings, you don't have to wait  for upstream support and can bind the Copy and Paste keycodes to the copy/paste actions in that app.
